http/src/main/java/com/github/awsjavakit/http/CachedTokenProvider.java

```diff
@@ -2,14 +2,15 @@
 
 import static java.util.Objects.isNull;
 
+import com.github.awsjavakit.http.token.OAuthTokenEntry;
 import java.time.Duration;
 import java.time.Instant;
 
 public class CachedTokenProvider implements TokenProvider {
 
   private final TokenProvider tokenProvider;
   private final Duration maxTokenAge;
-  private String token;
+  private OAuthTokenEntry token;
```

Comment:
The OAuthTokenEntry is a class that stores token information locally or in the cloud